shared/js/textChatStore.js

```diff
@@ -148,13 +155,24 @@ loop.store.TextChatStore = (function() {
     },
 
     /**
-     * Handles sending of a chat message.
+     * Handles sending of a chat message.  Adds a name (XXX yuck, this
+     * shouldn't be so side-effecty)
      *
      * @param {sharedActions.SendTextChatMessage} actionData
      */
     sendTextChatMessage: function(actionData) {
-      this._appendTextChatMessage(CHAT_MESSAGE_TYPES.SENT, actionData);
-      this._dataDriver.sendTextChatMessage(actionData);
+      var messageToSend = {};
+
+      if (actionData.contentType === CHAT_CONTENT_TYPES.TEXT) {
+        // text chat types get the sender's name added
+        Object.assign(messageToSend, actionData,
+          { displayName: this.getStoreState("displayName") }); // XXX no test cov
+      } else {
+        Object.assign(messageToSend, actionData);
+      }
```

Comment:
This might be a bit simpler:
```js
let messageToSend = Object.assign({}, actionData);
if (..) {
  Object.assign(messageToSend, { displayName: getStore(..) });
}
```